src/XRP/model/xrp-deposit-preauth.ts

```diff
@@ -42,6 +50,11 @@ export default class XrpDepositPreauth {
         undefined,
         xrplNetwork == XrplNetwork.Test || xrplNetwork == XrplNetwork.Dev,
       )
+    } else {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'DepositPreauth protobuf provides neither `Authorize` nor `Unauthorize` fields.',
+      )
```

Comment:
this seems reasonable.

if you want...i would also be down if you refactored the logic here. something like

```
if (authorize) {
  const authorizeXAddress = the thing it does
  return XrpDepositPreauth(authorizeXAddress, undefined)
}

if (unauthorize) {
  const unauthorizeXAddress = the thing it does
  return new XrpDepositPreauth(undefined, unauthorizeXAddress)
}

throw new XrpError(XrpErrorType.MalformedProtobuf
```

this tightens up variable space and simplifies the branching. totally up to you though, as i know you didn't write the original logic!